web/html/src/components/input/InputBase.tsx

```diff
@@ -173,7 +177,10 @@ export class InputBase<ValueType = string> extends React.Component<InputBaseProp
     }
 
     if (!this.props.disabled && (value || this.props.required)) {
-      if (this.props.required && !value) {
+      const noValue = Array.isArray(this.props.name) ?
+        value == null || Object.values(value).filter(v => v).length === 0 :
+        !value;
```

Comment:
Similar to the other comments above, it would be better to be explicit with the intent here, perhaps reusing the `isEmptyValue` idea.

Reply:
OK, I then changed the expression to get rid of the ternary operator:
```ts
_isNil(value) || (Array.isArray(this.props.name) && Object.values(value).filter(v => !this.isEmptyValue(v)).length === 0)
```